elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -160,6 +161,76 @@ describe('peerPool', () => {
 		});
 	});
 
+	describe('#request', () => {
+		let caughtError: Error;
+		beforeEach(async () => {
+			(peerPool['_peerSelectForRequest'] as any) = sandbox
+				.stub()
+				.returns([] as ReadonlyArray<P2PPeerInfo>);
+			try {
+				await peerPool.request({ procedure: 'proc', data: 123 });
+			} catch (err) {
+				caughtError = err;
+			}
+		});
+
+		it('should call _peerSelectForRequest with all the necessary options', async () => {
+			expect(peerPool['_peerSelectForRequest']).to.be.calledWith({
+				peers: [],
+				nodeInfo: peerPool.nodeInfo,
+				peerLimit: 1,
+				requestPacket: { procedure: 'proc', data: 123 },
+			});
+		});
+
+		it('should throw an error if no peers are found', async () => {
```

Comment:
I miss an extra case when no error is thrown.

Reply:
This is outside the scope of this issue. This issue is about `nodeInfo` being absent from the function. We have a different issue already open about unit tests for `PeerPool` and it should be addressed as part of that issue.

Reply:
I suggest to add a test skeleton now to not to forget the case. We should do this kind of things frequently.